web/docs/database/operations/actions.md

```diff
@@ -0,0 +1,889 @@
+
+// ...
+
+const newTask = await createTask({ description: "Learn TypeScript" });
+await markTasAsDone({ id: 1 });
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```typescript
+import createTask from "@wasp/actions/createTask.js";
+import markTasAsDone from "@wasp/actions/markTasAsDone.js";
+
+// TypeScript automatically infers the return values and type-checks
+// the payloads.
+const newTask = await createTask({ description: "Keep learning TypeScript" });
+await markTasAsDone({ id: 1 });
+```
+
+</TabItem>
+</Tabs>
+
+:::caution
+Wasp will not stop you from importing a Action's NodeJS implementation from `./actions.{js,ts}` and calling it directly.
+
+However, it's best if you avoid doing so. Besides losing all useful features a Wasp Action provides (e.g., Entity injection), the raw Action will expect you to provide a context as a second argument (which is cumbersome).
+:::
+
+When using Actions on the client, you'll most likely want to use them inside a component:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx {4,25} title=src/client/pages/Task.jsx
+import React from "react";
+import { useQuery } from "@wasp/queries";
+import getTask from "@wasp/queries/getTask";
+import markTaskAsDone from "@wasp/actions/markTaskAsDone";
+
+export const TaskPage = ({ id }) => {
+  const { data: task } = useQuery(getTask, { id });
+
+  if (!task) {
+    return <h1>"Loading"</h1>;
+  }
+
+  const { description, isDone } = task;
+  return (
+    <div>
+      <p>
+        <strong>Description: </strong>
+        {description}
+      </p>
+      <p>
+        <strong>Is done: </strong>
+        {isDone ? "Yes" : "No"}
+      </p>
+      {isDone || (
+        <button onClick={() => markTaskAsDone({ id })}>Mark as done.</button>
+      )}
+    </div>
+  );
+};
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx {4,25} title=src/client/pages/Task.tsx
+import React from "react";
+import { useQuery } from "@wasp/queries";
+import getTask from "@wasp/queries/getTask";
+import markTaskAsDone from "@wasp/actions/markTaskAsDone";
+
+export const TaskPage = ({ id }: { id: number }) => {
+  const { data: task } = useQuery(getTask, { id });
+
+  if (!task) {
+    return <h1>"Loading"</h1>;
+  }
+
+  const { description, isDone } = task;
+  return (
+    <div>
+      <p>
+        <strong>Description: </strong>
+        {description}
+      </p>
+      <p>
+        <strong>Is done: </strong>
+        {isDone ? "Yes" : "No"}
+      </p>
+      {isDone || (
+        <button onClick={() => markTaskAsDone({ id })}>Mark as done.</button>
+      )}
+    </div>
+  );
+};
+```
+
+</TabItem>
+</Tabs>
+
+Since Actions don't require reactivity, they are safe to use inside components without a hook. Still, Wasp provides comes with the `useAction` hook you can use to enhance actions. Read all about it in the [Options Reference](#options-reference).
+
+### Error Handling
```

Comment:
I would maybe put this section, using entities and Prisma Error Helpers into the `Overview` section (since they are the same for actions and queries). 

I did something similar with the Social Auth docs 🤷

Reply:
Discussed it, postponed it.